File: Python/Programmers/올바른 괄호.py
def solution(s):
    answer = True
    stack = []

    for i in range(len(s)):
        if stack:
            if stack[-1] == '(':
                if s[i] == ')':
                    stack.pop()
                else:
                    stack.append(s[i])
            else:
                stack.append(s[i])
        else:
            stack.append(s[i])

    # print(stack)
    if stack:
        answer = False

    return answer

File: Java/Programmers/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;

        int pair = 0;

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                pair += 1;
            } else {
                if (pair >= 1) {
                    pair -= 1;
                } else {
                    pair += 1;
                }
            }
        }

        if (pair != 0) {
            answer = false;
        }

        return answer;
    }
}